src/app/components/transaction/bitcoin/bitcoin-transaction.utils.ts

```diff
@@ -0,0 +1,42 @@
+import BigNumber from 'bignumber.js';
+import { truncateMiddle } from '@stacks/ui-utils';
+
+import { initBigNumber } from '@app/common/utils';
+
+import { BitcoinTransaction } from '@app/common/transactions/bitcoin/transaction.models';
+import { BTC_DECIMALS } from '@shared/constants';
+
+const satToBtc = (sat: number | string | BigNumber) => {
+  const satBigNumber = initBigNumber(sat);
+  return satBigNumber.shiftedBy(-BTC_DECIMALS);
+};
+
+export const getBitcoinTxCaption = (transaction: BitcoinTransaction) =>
+  truncateMiddle(transaction.txid, 4);
+
+// If vin array contains a prevout with a scriptpubkey_address equal to
+// the address, then that is the current address making a `Sent` tx (-)
+// and the value of the prevout is the tx amount
+const transactionsSentByAddress = (address: string, transaction: BitcoinTransaction) =>
+  transaction.vin.filter(input => input.prevout.scriptpubkey_address === address);
+
+// If vout array contains a scriptpubkey_address equal to the address,
+// then that is a `Receive` tx (+) and the value is the tx amount
+const transactionsReceivedByAddress = (address: string, transaction: BitcoinTransaction) =>
+  transaction.vout.filter(output => output.scriptpubkey_address === address);
+
+export function getBitcoinTxValue(address: string, transaction: BitcoinTransaction) {
+  const inputs = transactionsSentByAddress(address, transaction);
+  const outputs = transactionsReceivedByAddress(address, transaction);
+  const totalInputValue = satToBtc(
+    inputs.reduce((acc, vin) => acc + (vin.prevout.value || 0), 0)
+  ).toNumber();
+  const totalOutputValue = satToBtc(
+    outputs.reduce((acc, vout) => acc + (vout.value || 0), 0)
+  ).toNumber();
+  // This condition handles when change is sent back to the sender address
+  if (inputs.length && outputs.length) return '-' + (totalInputValue - totalOutputValue);
+  if (inputs.length) return '-' + totalInputValue;
+  if (outputs.length) return totalOutputValue;
+  return '';
```

Comment:
What type does this fn return currently? `number | string`? Could it just be a single type?